polyfactory/collection_extender.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import random
+from abc import ABC
+from typing import Any, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import GenericAlias  # type: ignore
+
+
+class CollectionExtender(ABC):
+    __types__: tuple[type, ...]
+
+    @staticmethod
+    def _extend_type_args(type_args: tuple[Any, ...], number_of_args: int) -> tuple[Any, ...]:
+        raise NotImplementedError
+
+    @classmethod
+    def _subclass_for_type(cls, annotation_alias: GenericAlias) -> type[CollectionExtender]:
+        for subclass in cls.__subclasses__():
+            try:
+                is_subclass = issubclass(annotation_alias.__origin__, subclass.__types__)
```

Comment:
Look at the helper we have to get safe subclass

Reply:
I'm not sure what you're referring to.

Reply:
This one I think:
https://github.com/litestar-org/polyfactory/blob/39eecb520e9704f3a8dd1593b2f91aaae13d046a/polyfactory/utils/predicates.py#L30